cli/src/main/java/com/devonfw/tools/ide/tool/yarn/Yarn.java

```diff
@@ -0,0 +1,121 @@
+package com.devonfw.tools.ide.tool.yarn;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.tool.ToolInstallation;
+import com.devonfw.tools.ide.tool.npm.Npm;
+import com.devonfw.tools.ide.tool.repository.ToolRepository;
+import com.devonfw.tools.ide.version.GenericVersionRange;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link ToolCommandlet} for <a href="https://yarnpkg.com">yarn</a>.
+ */
+public class Yarn extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Yarn(IdeContext context) {
+
+    super(context, "yarn", Set.of(Tag.JAVA_SCRIPT, Tag.BUILD));
+  }
+
+  @Override
+  protected boolean isIgnoreSoftwareRepo() {
+
+    return true;
+  }
+
+  @Override
+  public Path getToolPath() {
+
+    return this.context.getSoftwarePath().resolve("node");
+  }
+
+  @Override
+  public Path getToolBinPath() {
+
+    return getToolPath();
+  }
+
+  private boolean hasNodeBinary(String binary) {
+
+    Path toolPath = getToolPath();
+    Path yarnPath = toolPath.resolve(binary);
+    return Files.exists(yarnPath);
+  }
+
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    if (hasNodeBinary("yarn")) {
+      String version = this.context.newProcess().runAndGetSingleOutput("yarn", "-v");
+      return VersionIdentifier.of(version);
+    }
+    this.context.debug("Yarn is not installed in {}", getToolPath());
+    return null;
+  }
+
+  @Override
+  public String getInstalledEdition() {
+
+    if (hasNodeBinary("yarn")) {
+      return "yarn";
+    }
+    return null;
+  }
+
+  @Override
+  public ToolInstallation installTool(GenericVersionRange version, ProcessContext pc, String edition) {
+
+    ToolRepository toolRepository = getToolRepository();
+    VersionIdentifier configuredVersion = getConfiguredVersion();
+    // UrlUpdater required for Yarn...
+    // VersionIdentifier resolvedVersion = toolRepository.resolveVersion(this.tool, edition, version, this);
```

Comment:
```suggestion
```

Reply:
Same here. We actually need an UrlUpdater for Yarn at least to know the available version numbers.
AFAIK Yarn/node cannot handle version patterns `4.*` and for the parts where it can it has a different syntax so `*!` would be a different syntax in JS ecosystem.
Maybe I should have created this PR as a draft.
Actually I only stepped in because the issue turned out to be more complex than initially expected due to what @julia-cap figured out with Yarn Classic vs. Yarn Modern.

What a review could additionally point out is that there is no JUnit yet included for Yarn and that in some scenarios the current implementation does not work so it is actually buggy and therefore incomplete.